enginetest/queries/column_alias_queries.go

```diff
@@ -20,6 +20,56 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 )
 
+var BrokenAliasQueries = []QueryTest{
+	{
+		// The dual table's schema can conflict with aliases
+		//
+		Query:    `select "foo" as dummy, (select dummy)`,
+		Expected: []sql.Row{{"foo", "foo"}},
+	},
+	{
+		// This query crashes Dolt with "panic: unexpected type x"
+		// Seems likely to be coming from dual table's schema and dummy value.
+		// https://github.com/dolthub/dolt/issues/4256
+		Query:    `SELECT *, (select i union select i) as a from mytable;`,
+		Expected: []sql.Row{{1, "first row", 1}, {2, "second row", 2}, {3, "third row", 3}},
+	},
+	{
+		// This query actually works correctly in GMS, but not in Dolt
+		// Removing the second "as a" makes it work in Dolt. This seems to happen because GMS uses a map of
+		// aliases by name and the second a alias overwrites the one that should be used. GMS should not allow
+		// the subquery to see the second "as a" alias, since that expression should only see aliases defined
+		// in previous expressions (i.e. forward-references are disallowed).
+		Query:    `SELECT 1 as a, (select a) as a;`,
+		Expected: []sql.Row{{1, 1}},
+	},
+	{
+		// Fails with an unresolved *plan.Project node error
+		// The second Project in the union subquery doens't seem to get its alias reference resolved
+		Query:    `SELECT 1 as a, (select a union select a) as b;`,
```

Comment:
i would have thought this one passed, unless we aren't passing through the parent scope